src/lib/wallet/GoodWalletProvider.js

```diff
@@ -596,9 +603,10 @@ export const useSwitchTokenModal = (onDismiss = noop) => {
 export const useFixedDecimals = (token = 'G$', chainId = null) => {
   const { toDecimals } = useFormatToken(token)
   const asDecimals = number => toDecimals(number, chainId)
+  const isEth = nativeMainnetTokens.includes(token)
 
   return isNativeToken(token)
-    ? number => trimEnd(decimalsToFixed(asDecimals(number), 18), '0')
+    ? number => trimEnd(decimalsToFixed(asDecimals(number), isEth ? 4 : 18), '0')
```

Comment:
all native tokens should be formatted with 4 decimals, so just 4 have to be enough
pls remove isEth check and nativeMainnetTokens const